JP-2803: Propagate asn keywords to resample output products for level 2 processing

tapastro opened the pull request:
<!-- If this PR closes a JIRA ticket, make sure the title starts with the JIRA issue number,
for example JP-1234: <Fix a bug> -->
Resolves [JP-2803](https://jira.stsci.edu/browse/JP-2803)

<!-- If this PR closes a GitHub issue, reference it here by its number -->
Closes #

<!-- describe the changes comprising this PR here -->
This PR adds assignment of `model.meta.asn.pool_name` and `{...}.table_name` to level 2b i2d products - this required propagating these keywords through an intermediate ModelContainer in the resample step.

There was discussion in the JIRA ticket of propagating values from the first contained model if any ModelContainer was found have no assigned values, but I felt that this covers the issue more cleanly - if there is a need for other behavior, please let me know.

NB: If a user runs e.g. `strun calwebb_image2 some_rate.fits`, the i2d output will have values of 'singleton' for both of these keywords. If using a MAST-provided level2 asn, the values propagate as expected.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

hbushouse reviewed (approved):
Looks good to me.

Could resample_spec have the same issue for the s2d products created from a single input cal file?

hbushouse commented:
Don't forget to run regtests against the PR branch!

tapastro commented:
I have a Jenkins-RT run going, but I'll restart now that I've added a line to resample_spec - it looks like the table_name was being added in the style of taking the first contained model keyword values and assigning to the ModelContainer, so I've added a line for the pool_name as well.

tapastro commented:
Regressions show 31 failures, all of which are keyword additions due to the new code.

hbushouse commented:
So this is ready to merge, but I'm going to wait a bit until a regtest run for another recent merge is finished and we have a chance to update truth files for that one (so that the differences don't get mingled together).
